Adding measure of average time per run & simplified output between runs

dssgabriel opened the pull request:
## Description

This PR adds a measure of the average time taken per run, along with the standard deviation of this value. The output has also been changed to remove displays of `Run x/X time: x.xxx` between each run and avoid reprinting the available options in-between benchmarks. A helper function has been added in case the user wants to re-check what these options are. Lastly, the Makefile has been rewritten for something that is arguably cleaner.

## Motivation
### Average time measurement
With the previous implementation, and because of the program waits for user input, it was hard to gather statistics regarding the time taken per run. Instead of printing the latency each run, its value is stored in a static array which allows to compute a mean and standard deviation at the end of the benchmark. This simplifies the output for the user and helps it see at a glance if the benchmark is valid (a high value for the standard deviation would indicate instabilities and that the benchmark is probably not correct from a statistical stand point).

### `printHelp` function
This function was purely added to de-clutter the output in-between benchmarks (especially when the user wants to compare their results). The help for the available commands is only printed once (when the program is first started) and is then accessible to the user with the newly added option `11`.

### Makefile
The rewritten Makefile is more similar to what's standard, with macros to enhance re-usability and a `clean` target to remove object files and binaries.

## Screenshot
Here's a screenshot of how the program should now look with this PR:
![screenshot](https://user-images.githubusercontent.com/55387403/175170673-7328d5ca-3f3a-4ceb-99b5-9f0c78bb23e1.png)

## Note
The indentation of this PR seems to be completely off compared to the rest of the code. This may be because of a mismatch use of spaces and tabs. Should we consider using a tool such as `clang-format` to harmonize the formatting of the code?

dssgabriel committed:
Rewrote Makefile

dssgabriel committed:
Added average time measurement over multiple runs

- Removed timing displays between each run;
- Added compute of mean time per run (with standard deviation to validate the measures);
- Added `printHelp` function to avoid re-printing the available 'commands' each time.

dssgabriel committed:
Cleanup of object files + old Makefile

dssgabriel committed:
Enhanced output + added option for standard AVX

dssgabriel committed:
Added gitignore

dssgabriel committed:
Fixed score compute of AVX512/best FLOPS benchmark

FCLC reviewed (changes requested):
I'm looking at how `void currentFunction` is implemented, and it seems to cause/have issues with GCC-12+ due to non-predictive states that leave it as uninitialized.

Initializing the value to 0 cleans that up. 

Beyond that, great patch, as soon as that's done, LGTM!

dssgabriel committed:
Removed warnings: set `currentFunction` pointer to null

dssgabriel commented:
I changed the declaration of the `currentFunction` function pointer to be initialized to `nullptr`, this effectively removes the GCC-12.x warnings!

I also added a clang-format file in a previous commit which updated almost all lines of the code, I hope this is not too annoying :sweat_smile:

FCLC reviewed (approved):
Looks good to me!
